app/services/repository_actions/repository_cell_resolver.rb

```diff
@@ -0,0 +1,113 @@
+# frozen_string_literal: true
+
+module RepositoryActions
+  class RepositoryCellResolver
+    def initialize(cell, new_row, team)
+      @cell    = cell
+      @new_row = new_row
+      @team    = team
+    end
+
+    def call
+      case @cell.value_type
```

Comment:
You could probably omit this `case` statement with some ruby metaprogramming (generate String name of the function based on the `@cell.value_type`, and eval it). But I'm not sure if that is good or bad practice 😄.